tools/cldr/gencoll.js

```diff
@@ -72,79 +62,44 @@ process.argv.forEach(function (val, index, array) {
     }
 });
 
-if (process.argv.length < 4) {
+if (process.argv.length < 2) {
     console.error('Error: not enough arguments');
     usage();
 }
 
-iso15924FileName = process.argv[2];
-ucdDir = process.argv[3];
-cldrDir = process.argv[4];
-if (process.argv.length > 5) {
-    toDir = process.argv[5];
+if (process.argv.length > 2) {
+    toDir = process.argv[2];
 }
 
 console.log("gencoll - generate collation data.\n" +
-    "Copyright (c) 2014 JEDLSoft\n");
+    "Copyright © 2014, 2020 JEDLSoft\n");
 
-fs.exists(iso15924FileName, function (exists) {
-    if (!exists) {
-        console.error("Could not access file " + iso15924FileName);
-        usage();
-    }
-});
-fs.exists(ucdDir, function (exists) {
-    if (!exists) {
-        console.error("Could not access UCD directory " + ucdDir);
-        usage();
-    }
-});
-fs.exists(cldrDir, function (exists) {
-    if (!exists) {
-        console.error("Could not access CLDR directory " + cldrDir);
-        usage();
-    }
-});
-
-fs.exists(toDir, function (exists) {
-    if (!exists) {
-        console.error("Could not access target directory " + toDir);
-        usage();
-    }
-});
-
-scriptFileName = ucdDir + "/Scripts.txt";
-
-fs.exists(scriptFileName, function (exists) {
-    if (!exists) {
-        console.error("Could not access file " + scriptFileName);
-        usage();
-    }
-});
+if (!fs.existsSync(toDir)) {
+    console.error("Could not access target directory " + toDir);
+    usage();
+}
 
 ducetFileName = cldrDir + "/common/uca/allkeys_DUCET.txt";
```

Comment:
Because there is no npm module containing this file, couldn't be updated, right?

Reply:
Correct. Also, gencoll.js doesn't really work. I never finished it. Instead, I did the data files by hand. It was a pain in the you-know-what, but it was faster than trying to figure out the stupid ambiguous unicode file syntax.